lib/expublish/commitizen.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Expublish.Commitizen do
+  @moduledoc """
+  Implements commitizen specification.
+
+  https://www.conventionalcommits.org/en/v1.0.0/#specification
+  """
+
+  defstruct all: [], patch: [], feature: [], breaking: []
+
+  def run(commits) do
+    collect(commits)
+  end
+
+  defp collect(commits, acc \\ %__MODULE__{})
+
+  defp collect([], acc), do: acc
+
+  defp collect(["fix" <> _ = commit | rest], acc) do
+    collect(rest, %__MODULE__{acc | all: [commit | acc.all], patch: [commit | acc.patch]})
+  end
+
+  defp collect(["feat" <> _ = commit | rest], acc) do
+    collect(rest, %__MODULE__{acc | all: [commit | acc.all], feature: [commit | acc.feature]})
+  end
+
+  defp collect(["BREAKING CHANGE" <> _ = commit | rest], acc) do
```

Comment:
If I understand correctly the [specification](https://www.conventionalcommits.org/en/v1.0.0/#specification), `BREAKING CHANGE:` is not a valid type for a commit. That is, a commit subject cannot start with `BREAKING CHANGE: <description>`.

`BREAKING CHANGE` (or `BREAKING-CHANGE`) is a valid _token_ to be found in the footer.

If we want to indicate a breaking change by looking only at the first line of the commit, we should look for a `!`  after the commit type (or scope), like in `fix!: some fix that is also a breaking change` or `fix(parser)!: some fix to the parser that is a breaking change`

Reply:
A naïve attempt to be closer to the spec  wrt breaking changes(while still using `--oneline` for fetching commits): https://github.com/marc0s/expublish/commit/ecae0d082b1e12bda84bcc9e0cab87be4ef01c0d